src/managers/GuildEventManager.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const CachedManager = require('./CachedManager');
+const { TypeError, Error } = require('../errors');
+const GuildEvent = require('../structures/GuildEvent');
+const { PrivacyLevels, GuildEventEntityTypes } = require('../util/Constants');
+
+class GuildEventManager extends CachedManager {
+  constructor(guild, iterable) {
+    super(guild.client, GuildEvent, iterable);
+
+    /**
+     * The guild this manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of this manager
+   * @type {Collection<Snowflake, GuildEvent>}
+   * @name GuildEventManager#cache
+   */
+
+  /**
+   * Data that resolves to give a GuildEvent object. This can be:
+   * * A GuildEvent object
+   * * A Snowflake
+   * @typedef {GuildEvent|Snowflake} GuildEventResolvable
+   */
+
+  /**
+   * Options used to create a guild event.
+   * @typedef {Object} GuildEventCreateOptions
+   * @property {string} name The name of the event
+   * @property {Date} scheduledStartTime The time to schedule the event at
+   * @property {PrivacyLevel|number} privacyLevel The privacy level of the event
+   * @property {GuildEventEntityType|number} entityType The scheduled entity type of the event
+   * @property {string} [description] The description of the event
+   * @property {StageChannel|Snowflake} [channel] The stage channel of the event
```

Comment:
You could create a `StageChannelResolvable` `@typedef` for this, akin to others such as [`VoiceChannelResolvable`](https://discord.js.org/#/docs/main/main/typedef/VoiceChannelResolvable) and [TextChannelResolvable](https://discord.js.org/#/docs/main/main/typedef/TextChannelResolvable) etc.

Reply:
upstream changed these to just `channel id` now, so it won't be required here. This combination is referenced in a lot of places, so a new PR adding the typedef replacing all of them will be a better idea.

Reply:
> This combination is referenced in a lot of places

I see that now! I have opened #6568 then.